# An exploratory analysis of differential regulation patters of m6A on long non-coding and coding RNAs
## Authored By Feiyue Yang, Chase Armer, Claudia Aiello
### A Final Project for Deep Sequencing Fall 2023, Columbia University, taught by Dr. Peter Sims, Dr. Yufeng Shen, and Dr. Chaolin Zhang 

Welcome to the repository for our Deep Sequencing Final Project.

### Repository Structure
`code/` Contains all the code used in our analysis, organized by Methods Section<br>
`data/` Contains all the data used in our analysis <br>
`figures/` Contains all the figures generated in our analysis, organized by Figure Number

### Abstract
This study investigates the differential m6A methylation patterns and their regulatory impacts on long non-coding
RNAs (lncRNAs) and coding RNAs in mouse embryonic stem cells. The primary focus is on the role of m6A methyltrans-
ferases (writers), particularly METTL3, and m6A-binding proteins (readers), specifically YTHDF proteins, in modulating RNA
stability and expression. Our results reveal that knockout of METTL3 leads to a relative downregulation of lncRNAs compared
to protein-coding RNAs, with a notable variance in expression based on the number of m6A sites present in the genes. This
pattern is consistent across different m6A regulatory proteins, as evidenced by parallel observations in YTHDF1/2/3 triple-
knockout conditions. We also explore the m6A reading strategy differences between coding RNAs and lncRNAs, employing
integrated analysis of differential expression in METTL3-knockout and YTHDF1/2/3-triple-knockout datasets. Our findings
suggest that while there is a similarity in the total m6A reading manners of YTHDF proteins between coding RNAs and
lncRNAs, individual YTHDF proteins exhibit distinct non-redundant reading abilities, especially in lncRNAs. The study also
delves into the complex relationship between m6A modifications and the transcriptional resilience of genes, with a particular
emphasis on housekeeping genes. Our data indicate that m6A modifications impart a nuanced influence on gene expression
fidelity, suggesting the involvement of additional regulatory mechanisms beyond m6A writers and readers. This research pro-
vides novel insights into the epitranscriptomic regulation of gene expression and highlights the intricate interplay between m6A
modifications and RNA stability in both coding and non-coding RNA landscapes.
