Raw fastq files were downloaded from the European Nucleotide Archive(ENA). Quality control was conducted with FastQC and reads were trimmed using cutadapt (version 4.5) (parameters: -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -a AAAAAAAAAA -a TTTTTTTTTT --times 2 -q 20 -m 25). Reads were mapped to genome mm10 using STAR (version 2.7.9) (parameters: –sjdbOverhang 74 –alignEndsType EndToEnd –outFilterMismatchNoverLmax 0.05 –twopassMode Basic). Only uniquely mapped reads were used for downstream analysis. Sample counting was done using StringTie (version 2.2.1), quantifying mm10 Ensembl annotated genes.
